src/lib.rs

```diff
@@ -714,6 +714,95 @@ impl LiteSVM {
         }
     }
 
+    fn execute_transaction_readonly(&self, tx: VersionedTransaction) -> ExecutionResult {
+        let sanitized_tx = match self.sanitize_transaction(tx) {
+            Ok(s_tx) => s_tx,
+            Err(err) => {
+                return ExecutionResult {
+                    tx_result: Err(err),
+                    ..Default::default()
+                }
+            }
+        };
+        self.execute_sanitized_transaction_readonly(sanitized_tx)
+    }
+
+    fn execute_transaction_no_verify_readonly(&self, tx: VersionedTransaction) -> ExecutionResult {
+        let sanitized_tx = match self.sanitize_transaction_no_verify(tx) {
+            Ok(s_tx) => s_tx,
+            Err(err) => {
+                return ExecutionResult {
+                    tx_result: Err(err),
+                    ..Default::default()
+                };
+            }
+        };
+        self.execute_sanitized_transaction_readonly(sanitized_tx)
+    }
+
+    fn execute_sanitized_transaction_readonly(
```

Comment:
Mind extracting whatever helper functions are needed to reduce duplication in this function body?

Reply:
nvm i'll just take it now